### PR TITLE
ICU-20160 Add check for Python 3 in autoconf and nmake.

### PR DESCRIPTION
**Pull request by sffc**
##### Checklist

- [x] Issue filed at https://unicode-org.atlassian.net :  ICU-20160
- [x] Update PR title to include Issue number
- [x] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

**Comment by sffc** [comment]
@jefgen Suggestions on where to put this for the Windows build?

**Comment by jefgen** [comment]
@sffc I'm trying to think of a location that wouldn't break anyone... I think maybe one location might be in the file `makedata.mak` as an informational message only (so as to not break the build).

File:  https://github.com/unicode-org/icu/blob/master/icu4c/source/data/makedata.mak 

Perhaps adding something like this near the beginning of the file?
```
!MESSAGE Information: Notice: Python 3 not found. ICU 64 will require Python 3 to build. See ICU-10923 for more information.
```

File: icu4c/source/configure.ac [hunk 1]
```diff
@@ -1462,4 +1464,11 @@ then
         echo "## Expect build failures in the 'data', 'test', and other directories."
 fi
 
+if test -z "$PYTHON3";
+then
+    echo "##"
+    echo "## Notice: Python 3 not found. ICU 64 will require Python 3 to build. See ICU-10923."
```
Comment (srl295) [inline icu4c/source/configure.ac]: suggestion: 
- `ICU versions 64 and later will require…`
- make the ticket number have a URL (and include a link to Python on the ticket)

**Comment by sffc** [comment]
Added Windows and made changes requested by Steven.  Please take another look.

**Comment by sffc** [comment]
One more try on Windows.  It looks like `py -3` works but not `python3`, at least not based on the official Python 3 installation from https://www.python.org/downloads/.

**Comment by sffc** [comment]
@srl295 @jefgen Can I have approval to merge this in?

**Pull request merged**